modules/s3-bucket/main.tf

```
@@ -57,11 +87,11 @@ data "aws_iam_policy_document" "custom_policy" {
     actions = var.custom_policy_actions
 
     resources = [
-      format("arn:aws:s3:::%s-%s-%s-%s", module.this.namespace, module.this.environment, module.this.stage, module.this.name),
-      format("arn:aws:s3:::%s-%s-%s-%s/*", module.this.namespace, module.this.environment, module.this.stage, module.this.name)
+      format("arn:%s:s3:::%s", local.aws_partition, module.this.id),
+      format("arn:%s:s3:::%s/*", local.aws_partition, module.this.id)
     ]
     principals {
-      identifiers = local.custom_policy_account_arns
+      identifiers = length(local.custom_policy_account_arns) > 0 ? local.custom_policy_account_arns : ["*"]
```

I do not like the default being fully open access. That is dangerous. Default should be an error, because it is easy enough to specify `["*"]` if that is what you want.
```suggestion
      identifiers = local.custom_policy_account_arns
```